### PR TITLE
fix: nightly bug fixes 2026-06-19

### PR DESCRIPTION
**Pull request by umairnadeem**
## Summary
- Add strict route-boundary validation for `/api/v1/tts/voices` search parameters.
- Reject malformed voice search query values before calling the TTS provider instead of silently dropping invalid filters.
- Keep permissive metadata parsing separate from API request validation.

## Bugs fixed
- `limit=abc` (and other malformed positive-integer limits) previously parsed to `undefined`, causing the route to return an unbounded/default voice search instead of a client error.
- Invalid voice search trait values now fail request validation at the route boundary rather than being silently ignored.

## Tests
- Updated `GET /api/v1/tts/voices` coverage to assert valid `limit` values are parsed and forwarded as numbers.
- Added regression coverage that invalid voice search parameters return `400` and do not call the provider.

## Validation
- `npm test -- app/api/v1/__tests__/routes.test.ts` ✅
- `npm test -- --passWithNoTests` ✅
- `npm run build` ✅
- `npm run lint` ✅
- `npm run format:check` ✅
- `npm run typecheck` ✅
- `npm run knip` ✅
- `npm run test:coverage` ✅

## Open PR overlap check
Considered open PRs #309, #305, #304, #303, and #302. This PR changes only:
- `app/api/v1/__tests__/routes.test.ts`
- `app/api/v1/tts/voices/route.ts`
- `lib/project/types.ts`

These files and the voice-search route-boundary validation theme do not overlap with the open dropdown, project rehydrate, Playwright smoke, connector architecture, or tooltip-maintainability PRs.

## Notes
- Claude Code was invoked in the required YOLO print-mode style for the core pass, but it timed out without leaving a diff; Hermes completed the focused fallback fix and independently validated it.

### CODE REVIEW

File: lib/project/types.ts [hunk 1]
```diff
@@ -31,10 +37,19 @@ export const MetadataVoiceSchema = voiceTraitsSchema.extend({
 	resolvedVoiceId: optionalString,
 });
 
-export const voiceSearchParamsSchema = voiceTraitsSchema.extend({
-	query: optionalString,
-	name: optionalString,
-	limit: z.coerce.number().int().positive().optional().catch(undefined),
+const voiceSearchTraitsSchema = z.object({
```
Comment (umairnadeem) [inline lib/project/types.ts:40]: split-brain: `voiceSearchTraitsSchema` re-declares the exact same 6 trait fields as `voiceTraitsSchema` above, the only diff is strict vs `.catch(undefined)`. add a 7th trait enum later (emotion, whatever) and you have to remember to touch both objects or they silently drift. could derive both from one base so there's a single source of truth, e.g. keep a strict trait object as canonical and build the permissive metadata variant off it. not blocking but it's the kind of dupe that quietly rots.

File: lib/project/types.ts [hunk 2]
```diff
@@ -10,6 +10,12 @@ import type { AspectRatio } from "@/lib/video/aspectRatio";
 import type { TransitionType } from "@/lib/video/transitions";
 
 const optionalString = z.string().min(1).optional().catch(undefined);
+const voiceSearchString = z.string().min(1).optional();
```
Comment (umairnadeem) [inline lib/project/types.ts:13]: footgun: `voiceSearchString` is `min(1)` with no `.catch`, so `?query=` (empty string, a super common shape when a client serializes an unset filter) now 400s instead of being treated as 'no query filter'. old schema caught it -> undefined. same hits `name`/`description`. if empty-means-absent is the intended UX, `.transform(s => s || undefined)` (or dropping min(1) for these search fields) keeps the old behavior while still rejecting genuinely bad input. up to you whether that empty-param shape is real for this API.

**Review by umairnadeem — COMMENTED** [review]
tightening the route boundary is the right move. swapping the silent `.catch(undefined)` parse for `safeParse` + `badRequest` so `limit=abc` returns a 400 instead of a default unbounded search is a real fix, and the regression test asserting the provider isn't called is exactly what i'd want. two things worth a look below (one maintainability, one edge case), neither blocking.

**Comment by umairnadeem** [comment]
useless
